purchase_order_line_schedule_manual_delivery/wizard/create_manual_stock_picking.py

```diff
@@ -0,0 +1,167 @@
+    _name = "create.stock.picking.wizard.schedule.line"
+    _description = "Create Manual Stock Picking Wizard Line"
+
+    wizard_id = fields.Many2one(
+        string="Wizard",
+        comodel_name="create.stock.picking.wizard",
+        required=True,
+        ondelete="cascade",
+    )
+    schedule_line_id = fields.Many2one("purchase.order.line.schedule")
+    purchase_order_line_id = fields.Many2one(
+        comodel_name="purchase.order.line", related="schedule_line_id.order_line_id"
+    )
+    name = fields.Text(string="Description", readonly=True)
+    product_id = fields.Many2one(
+        "product.product",
+        related="purchase_order_line_id.product_id",
+        string="Product",
+    )
+    product_uom = fields.Many2one(
+        "uom.uom",
+        related="purchase_order_line_id.product_uom",
+        string="Unit of Measure",
+    )
+    date_planned = fields.Datetime(related="schedule_line_id.date_planned")
+    product_qty = fields.Float(
+        string="Quantity",
+        related="schedule_line_id.product_qty",
+        digits="Product Unit of Measure",
+    )
+    existing_qty = fields.Float(related="schedule_line_id.existing_qty",)
+    remaining_qty = fields.Float(
+        string="Remaining Quantity",
+        compute="_compute_remaining_qty",
+        readonly=True,
+        digits="Product Unit of Measure",
+    )
+    qty = fields.Float(
+        string="Receive",
+        digits="Product Unit of Measure",
+        help="This is the quantity taken into account to create the picking",
+    )
+    price_unit = fields.Float(
+        related="purchase_order_line_id.price_unit", readonly=True
+    )
+    currency_id = fields.Many2one(
+        "res.currency", related="purchase_order_line_id.currency_id"
+    )
+    partner_id = fields.Many2one(
+        "res.partner", related="purchase_order_line_id.partner_id", string="Vendor",
+    )
+    taxes_id = fields.Many2many(
+        "account.tax", related="purchase_order_line_id.taxes_id"
+    )
+
+    def _compute_remaining_qty(self):
+        for line in self:
+            line.remaining_qty = line.product_qty - line.existing_qty
+
+    def _prepare_stock_moves(self, picking):
+        po_line = self.purchase_order_line_id
+        return po_line._prepare_stock_moves(picking)
+
+    def _create_stock_moves(self, picking):
+        values = []
+        for line in self:
+            for val in line.with_context(schedule_line_id=line.id)._prepare_stock_moves(
+                picking
+            ):
+                if val.get("product_uom_qty", False):
+                    val["product_uom_qty"] = line.product_uom._compute_quantity(
+                        line.qty, line.product_uom, rounding_method="HALF-UP"
+                    )
+                if (
+                    val.get("location_dest_id", False)
+                    and not line.wizard_id.picking_id
+                    and line.wizard_id.location_dest_id
```

Comment:
Negative quantities are allowed in schedules but will generate normal incoming shipments.

Reply:
Disallowing negative quantities would be better I think.